File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/jdbc3/JDBC3Savepoint.java
package io.roastedroot.sqlite4j.jdbc3;

import java.sql.SQLException;
import java.sql.Savepoint;
import org.jspecify.annotations.NonNull;
import org.jspecify.annotations.Nullable;

public class JDBC3Savepoint implements Savepoint {

    final int id;

    final @Nullable String name;

    JDBC3Savepoint(int id) {
        this.id = id;
        this.name = null;
    }

    JDBC3Savepoint(int id, @NonNull String name) {
        this.id = id;
        this.name = name;
    }

    public int getSavepointId() throws SQLException {
        return id;
    }

    public String getSavepointName() throws SQLException {
        return name == null ? String.format("SQLITE_SAVEPOINT_%s", id) : name;
    }
}

File: modules/sqlite-jdbc/src/main/kotlin/org/example/sqlite/jdbc/jdbc3/JDBC3Savepoint.kt
package org.example.sqlite.jdbc.jdbc3

import java.sql.SQLException
import java.sql.Savepoint

class JDBC3Savepoint : Savepoint {
    val id: Int

    val name: String?

    internal constructor(id: Int) {
        this.id = id
        this.name = null
    }

    internal constructor(id: Int, name: String) {
        this.id = id
        this.name = name
    }

    @Throws(SQLException::class)
    override fun getSavepointId(): Int {
        return id
    }

    @Throws(SQLException::class)
    override fun getSavepointName(): String {
        return if (name == null) String.format("SQLITE_SAVEPOINT_%s", id) else name
    }
}
